examples/question_answering/question_answering/seed_test_suite.py

```diff
@@ -0,0 +1,162 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+from argparse import ArgumentParser
+from argparse import Namespace
+from collections import defaultdict
+from typing import List
+from typing import Tuple
+
+import pandas as pd
+from question_answering.workflow import GroundTruth
+from question_answering.workflow import TestCase
+from question_answering.workflow import TestSample
+from question_answering.workflow import TestSuite
+from utils import normalize_string
+
+import kolena
+from kolena.workflow.annotation import ClassificationLabel
+
+
+BUCKET = "kolena-public-datasets"
+DATASET = "CoQA"
+
+
+def clean_name(name: str) -> str:
+    return name[9:] if name.startswith("metadata_") else name
+
+
+def create_test_suite_by_question(dataset: TestCase, data: List[Tuple[TestSample, GroundTruth]]) -> None:
+    question_types = ["what", "who", "how", "did", "where", "was", "when", "is", "why", "other"]
+    samples_by_question_type = {value: [] for value in question_types}
+
+    # Organize samples into the dictionary
+    for ts, gt in data:
+        question_type = ts.metadata["type_of_question"]
+        if question_type in samples_by_question_type:
+            samples_by_question_type[question_type].append((ts, gt))
+        else:
+            samples_by_question_type["other"].append((ts, gt))
+
+    # Create test cases using the organized dictionary
+    test_cases = TestCase.init_many(
+        data=[
+            (f"question type :: {name} :: {DATASET}", samples_by_question_type.get(name, [])) for name in question_types
+        ],
+        reset=True,
+    )
+
+    test_suite = TestSuite(
+        f"question types :: {DATASET}",
+        test_cases=[dataset, *test_cases],
+        reset=True,
+    )
+    print(f"created test suite: {test_suite.name} v{test_suite.version}")
+
+
+def create_test_suite_by_conversation_length(dataset: TestCase, data: List[Tuple[TestSample, GroundTruth]]) -> None:
+    depths = sorted({test_sample.turn for test_sample, _ in data})
+    samples_by_turn = {value: [] for value in depths}
+
+    # Organize samples into the dictionary
+    for ts, gt in data:
+        samples_by_turn[ts.turn].append((ts, gt))
+
+    # Create test cases using the organized dictionary
+    test_cases = TestCase.init_many(
+        data=[
+            (f"conversation depth :: {number:02d} :: {DATASET}", samples_by_turn[number])
+            for number in depths
+            if len(samples_by_turn[number]) > 10
```

Comment:
Are we intentionally skipping small sized test cases (i.e. > 10)? How is this number chosen?

Reply:
I intentionally skipped the small sized test cases (there were only 4 samples in them) for a clean number of 20 test cases + 1 complete test case.

Reply:
hmm yeah I think maybe adding that as a comment somewhere could be helpful for anyone who's going through this example.